DataFormats/BTauReco/src/TaggingVariable.cc

```diff
@@ -127,7 +127,7 @@ const char* const TaggingVariableTokens[] = {
   /* [trackPtRatio]                             = */ "trackPtRatio",
   /* [trackPParRatio]                           = */ "trackPParRatio",
 
-  /* [trackIp2dVal]                             = */ "trackIp2dVal",
+  /* [trackIp2dSig]                             = */ "trackIp2dSig",
```

Comment:
google for "map vs vector C++" and you will find all possible arguments (unfortunately many quite confusing)
essentially nodes in maps are allocated  on the heap individually and have memory overhead (list list) due to pointers between nodes.
vector has no memory overhead and has contiguous storage.
for simple payloads (float, int, small struct) it is pretty obvious that binary_search in a sorted vector will be faster
than map::find and will use much less space...
For larger payload caching issues will play a role... 
For string is even more complex as string are stored on the heap and therefore there is already the cost of sparse storage and cache misses
For string in particular hashing is very efficient: so unorder_map is most probably the best choice.
Actually for a finite (i..e known in advance) number of strings one can use 64bit hashing, verify that is unique and then use a double vector (ordered hashes and corresponding string),  search the hashing return the string at the corresponding location.
This is of course an over-killing for your application...
So just use unordered_map and verify that is NOT a bottle-neck